armory/scenarios/poison.py

```diff
@@ -225,47 +225,59 @@ def filter_dataset(self):
             and not self.check_run
         ):
             defense_config = copy.deepcopy(self.config["defense"] or {})
-            if "data_augmentation" in defense_config:
-                defense_config.pop("data_augmentation")  # NOTE: RESISC10 ONLY
 
-            # Assumes classifier_for_defense and classifier use same preprocessing function
-            defense_model_config = adhoc_config.get(
-                "defense_model", self.config["model"]
-            )
-            classifier_for_defense, _ = config_loading.load_model(defense_model_config)
-            log.info(
-                f"Fitting model {defense_model_config['module']}.{defense_model_config['name']} "
-                f"for defense {defense_config['name']}..."
-            )
-            # Flag to determine whether defense_classifier is trained directly
-            #     (default API) or is trained as part of detect_poisons method
-            if adhoc_config.get("fit_defense_classifier_outside_defense", True):
-                classifier_for_defense.fit(
+            if (
```

Comment:
Would it be possible to create a simple `PerfectFilter` module in `armory.art_experimental.poison_detection` that uses the same interface as the other poison filter defenses? If so, this would seem cleaner than checking for a specific kwarg in the scenario code

Reply:
Look at [5747c50](https://github.com/twosixlabs/armory/pull/1484/commits/5747c5041f548e4a94af58511514c4406ad78280) to see what I had before

Reply:
if we do keep the check, I think
```
if defense_config["kwargs"].get("perfect_filter", False):
```
would be cleaner than 

```
 if (
                "perfect_filter" in defense_config["kwargs"]
                and defense_config["kwargs"]["perfect_filter"] is True
            )
```

Reply:
The False is unnecessary, as `.get("perfect_filter")` will default to None, which is false.

Reply:
I see, @swsuggs disregard my comment then. I suppose there's not a great way to propagate that info to the defense class without adding another check for a specific value